worldedit-core/src/main/java/com/fastasyncworldedit/core/extent/clipboard/CPUOptimizedClipboard.java

```diff
@@ -217,6 +219,17 @@ public Entity createEntity(Location location, BaseEntity entity) {
         return ret;
     }
 
+    @Nullable
+    @Override
+    public Entity createEntity(Location location, BaseEntity entity, UUID uuid) {
+        Map<String, Tag> map = new HashMap<>(entity.getNbtData().getValue());
+        NBTUtils.addUUIDToMap(map, uuid);
+        entity.setNbtData(new CompoundTag(map));
+        BlockArrayClipboard.ClipboardEntity ret = new BlockArrayClipboard.ClipboardEntity(location, entity);
+        entities.add(ret);
+        return ret;
```

Comment:
For ease of maintainability, how about refactoring this into a protected method? It's the same code in the CPUOP, DOC and MOC.